File: CodeWars/Lazy_Lewin/sol.py
def cost(s):
   cst = 0
   for c in s:
      if c >= 'a':
         cst += ord(c) - ord('a') + 1
      else:
         cst += ord(c) - ord('A') + 27
   return cst

n, k = map(int, input().split())

pre = [0] * (n + 1)

for i in range(n):
   s = input()
   pre[i + 1] = pre[i] + cost(s)

ans = int(10**9)
for i in range(k, n + 1):
   ans = min(ans, pre[i] - pre[i - k])

print(ans)

File: CodeWars/Lazy_Lewin/sol.cpp
#include <bits/stdc++.h>
using namespace std;

int cost(string s) {
    int cst = 0;
    for (char c : s) {
        if (c >= 'a') {
            cst += c - 'a' + 1;
        } else {
            cst += c - 'A' + 27;
        }
    }
    return cst;
}

int32_t main(){
    int n, k;
    cin >> n >> k;
    vector<int> cst(n);
    vector<int> pre(n + 1);
    for (int i = 0; i < n; ++i) {
        string s;
        cin >> s;
        cst[i] = cost(s);
        pre[i + 1] = pre[i] + cst[i];
    }
    int ans = 1e9;
    for (int i = k; i <= n; ++i) {
        ans = min(ans, pre[i] - pre[i - k]);
    }
    cout << ans << '\n';
    return 0;
}